core/app/models/spree/order.rb

```diff
@@ -450,7 +450,7 @@ def empty!
       line_items.destroy_all
       adjustments.destroy_all
       shipments.destroy_all
-      order_promotions.destroy_all
+      Spree::Bus.publish :order_emptied, order: self
```

Comment:
I can't recall exactly how it works.. is this always sync? Worried that the subscriber could be executed after the order is already recalculated.

Reply:
There's an ActiveJob and Sidekiq adapter for subscribers. If those are used in the subscriber, sync can't be guaranteed. However, the particular subscriber I'm adding here uses the `method` adapter, which is synchronous.

Reply:
Ok, nice! I didn't remember that the adapter configuration was set at the subscriber level.